File: src/main/java/com/ivaneye/intellijweb2/service/OrdersService.java
package com.ivaneye.intellijweb2.service;

import com.ivaneye.intellijweb2.model.Orders;
import com.ivaneye.intellijweb2.respository.OrdersRespository;
import com.ivaneye.intellijweb2.param.OrdersParam;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

/**
 * <p>
 *  服务实现类
 * </p>
 *
 * @author wangyifan
 * @since 2018-02-12
 */
@Service
@Transactional
public class OrdersService {

    @Autowired
    private OrdersRespository ordersRespository;

    public Orders find(Long recId){
        return ordersRespository.selectByPrimaryKey(recId);
    }

    public Long create(OrdersParam param) {
        ordersRespository.insertAndGet(param);
        return param.getRecId();
    }

}

File: src/main/kotlin/com/ivaneye/intellijweb2/service/OrdersService.kt
package com.ivaneye.intellijweb2.service

import com.ivaneye.intellijweb2.controller.TestController
import com.ivaneye.intellijweb2.mapper.OrdersMapper
import com.ivaneye.intellijweb2.model.Orders
import com.ivaneye.intellijweb2.param.OrdersParam
import com.ivaneye.intellijweb2.respository.OrdersRespository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import javax.annotation.Resource

/**
 * Created by wangyifan on 18-2-19.
 */
@Service
@Transactional
open class OrdersService {

    @Autowired
    open lateinit var ordersRespository: OrdersRespository

    fun find(recId: Long): Orders {
        return ordersRespository!!.selectByPrimaryKey(recId)
    }

    fun create(param: OrdersParam): Long? {
        ordersRespository!!.insertAndGet(param)
        return param.recId
    }

}